src/utils/subscription-registry.js

```diff
@@ -111,18 +117,94 @@ class SubscriptionRegistry {
 	 * @returns {void}
 	 */
 	sendToSubscribers( name, msgString, sender ) {
-		if( this._subscriptions[ name ] === undefined ) {
+		function onTimeout( delayedBroadcasts, that ) {
+			let sockets = that._subscriptions[ delayedBroadcasts.name ];
+			if ( sockets ) {
+				delayedBroadcasts.uniqueSendersVector.sort( (a, b) => {
+					if ( a.sender < b.sender ) {
+						return -1;
+					} else if ( a.sender > b.sender ) {
+						return 1;
+					} else {
+						return 0;
+					}
+				} );
+
+				for ( let uniqueSender of delayedBroadcasts.uniqueSendersVector ) {
+					uniqueSender.message = delayedBroadcasts.sharedMessages.
+						substring( 0, uniqueSender.gaps[ 0 ].start );
+					let lastStop = uniqueSender.gaps[ 0 ].stop;
+					for ( let j = 1; j < uniqueSender.gaps.length; j++ ) {
+						uniqueSender.message += delayedBroadcasts.sharedMessages.
+							substring( lastStop, uniqueSender.gaps[ j ].start );
+						lastStop = uniqueSender.gaps[ j ].stop;
+					}
+					uniqueSender.message += delayedBroadcasts.sharedMessages.
+						substring( lastStop, delayedBroadcasts.sharedMessages.length );
+				}
+
+				let preparedMessage = SocketWrapper.prepareMessage( delayedBroadcasts.sharedMessages );
+				let j = 0;
+				for ( let socket of sockets ) {
+					if ( j < delayedBroadcasts.uniqueSendersVector.length &&
+						delayedBroadcasts.uniqueSendersVector[ j ].sender === socket.uuid ) {
+						if ( delayedBroadcasts.uniqueSendersVector[ j ].message.length ) {
+							socket.sendNative( delayedBroadcasts.uniqueSendersVector[ j ].message );
+						}
+						j++;
+					} else {
+						socket.sendPrepared( preparedMessage );
+					}
+				}
+				SocketWrapper.finalizeMessage( preparedMessage );
+			}
+
+			delayedBroadcasts.sharedMessages = '';
+			delayedBroadcasts.uniqueSendersVector = [];
+			delayedBroadcasts.uniqueSendersMap = {};
+			delayedBroadcasts.timer = null;
+		}
+
+		if (this._subscriptions[ name ] == undefined) {
 			return;
 		}
 
-		var i, l = this._subscriptions[ name ].length;
+		// todo: this should not have to be corrected here! fix where the real issue is!
```

Comment:
Some messages are invalid when they are passed to be sent, so some of them get the end char appended here.